nasa-gibs/worldview
# Improve file fetching at build time

## Description

>The time it takes to fetch colormaps, vector styles, vector data and layer-metadata increased significantly from the previous python code. This is because node has to async/await each file that it is fetching; we should investigate using promise.all and synchronous fetching to speed these tasks up.

>These changes would be made in the getCapabilities.js and getVisMetadata.js files.

>Alternatively, for layer-metadata files, GIBS could pre-combine all the layers into a all.json, hosted at their endpoint, allowing us to only need to fetch 1 single layer-metadata file. There has been some discussion to make this happen due to earthdata search also wanting a single file.

## How to test

1. `git checkout WV-2613-improve-file-fetching`
2. `npm ci`
3. `npm run build`
4. Confirm performance improvements
